### PR TITLE
ci: os update

### PR DESCRIPTION
**Pull request by crazy-max**

### CODE REVIEW

**Comment by crazy-max** [comment]
I see we still have an issue with bookworm but this time not related to #95: https://github.com/tonistiigi/xx/actions/runs/5497434314/jobs/10018164421#step:5:401

```
#30 [test  8/12] RUN --mount=type=cache,target=/pkg-cache     [ "${TEST_CMDS:-apt}" = "${TEST_CMDS#*apt}" ] && exit 0; [ ! -f /etc/debian_version ] || ./test-apt.bats
#30 0.099 1..12
#30 0.201 ok 1 no_cmd
#30 2.863 ok 2 native
#30 4.871 not ok 3 essentials
#30 4.872 # (from function `assert_success' in file bats-assert/src/assert.bash, line 114,
#30 4.873 #  in test file test-apt.bats, line 30)
#30 4.874 #   `assert_success' failed
#30 4.875 #
#30 4.876 # -- command failed --
#30 4.876 # status : 100
#30 4.876 # output (9 lines):
#30 4.876 #   Hit:1 http://deb.debian.org/debian bookworm InRelease
#30 4.877 #   Hit:2 http://deb.debian.org/debian bookworm-updates InRelease
#30 4.877 #   Hit:3 http://deb.debian.org/debian-security bookworm-security InRelease
#30 4.877 #   Reading package lists...
#30 4.877 #   + apt  show libstdc++-8-dev
#30 4.877 #
#30 4.878 #   WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
#30 4.878 #
#30 4.878 #   E: No packages found
#30 4.879 # --
#30 4.879 #
#30 4.906 ok 4 amd64 # skip
#30 9.282 ok 5 arm64
#30 15.02 ok 6 arm
#30 22.36 ok 7 armv6
#30 31.15 ok 8 s390x
#30 42.57 ok 9 ppc64le
#30 56.34 ok 10 386
#30 57.19 ok 11 skip-nolinux
#30 58.90 ok 12 checkpkg
```

#99 should fix this issue.

File: base/test-clang.bats [hunk 1]
```diff
@@ -79,6 +79,14 @@ testBuildHello() {
...
   export XX_NO_DOWNLOAD=1
   if [ -f /etc/debian_version ]; then skip; fi # clang has dependency on ld in debian
+  if [ -f /etc/alpine-release ]; then
+    # clang (since 16) has a dependency on linker tools in alpine edge and > 3.17
+    # https://github.com/alpinelinux/aports/blob/19182e91254966beefa20f8f110fd522c9f3454c/main/clang16/APKBUILD#L128
+    alpineRelease=$(cat /etc/alpine-release)
+    if [ "$alpineRelease" = "edge" ] || [ "$(xx-semver $alpineRelease 3.17.999)" = "1" ]; then
+      skip
+    fi
+  fi
```
Comment (crazy-max) [inline base/test-clang.bats:89]: Fixes nolinker test for alpine edge and > 3.17.

File: base/test-clang.bats [hunk 2]
```diff
@@ -79,6 +79,14 @@ testBuildHello() {
...
   export XX_NO_DOWNLOAD=1
   if [ -f /etc/debian_version ]; then skip; fi # clang has dependency on ld in debian
+  if [ -f /etc/alpine-release ]; then
+    # clang (since 16) has a dependency on linker tools since alpine 3.18
+    # https://github.com/alpinelinux/aports/blob/19182e91254966beefa20f8f110fd522c9f3454c/main/clang16/APKBUILD#L128
+    alpineRelease=$(cat /etc/alpine-release)
+    if grep PRETTY_NAME /etc/os-release | cut -d '=' -f 2 | tr -d '"' | grep -q "edge$" || [ "$(semver compare "$alpineRelease" "3.18")" -ge 0 ]; then
+      skip
+    fi
+  fi
```
Comment (crazy-max) [inline base/test-clang.bats:89]: Fixes nolinker test for alpine > 3.17.

File: base/test-clang.bats [hunk 3]
```diff
@@ -79,6 +79,14 @@ testBuildHello() {
...
   export XX_NO_DOWNLOAD=1
   if [ -f /etc/debian_version ]; then skip; fi # clang has dependency on ld in debian
+  if [ -f /etc/alpine-release ]; then
+    # clang (since 16) has a dependency on linker tools since alpine 3.18
+    # https://github.com/alpinelinux/aports/blob/19182e91254966beefa20f8f110fd522c9f3454c/main/clang16/APKBUILD#L128
+    alpineRelease=$(cat /etc/alpine-release)
+    if grep PRETTY_NAME /etc/os-release | cut -d '=' -f 2 | tr -d '"' | grep -q "edge$" || [ "$(semver compare "$alpineRelease" "3.17.9999")" -ge 0 ]; then
+      skip
+    fi
+  fi
```
Comment (crazy-max) [inline base/test-clang.bats:89]: Fixes nolinker test for alpine > 3.17.

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**